packages/visibility_detector/lib/src/visibility_detector_layer.dart

```diff
@@ -135,6 +135,8 @@ class VisibilityDetectorLayer extends ContainerLayer {
         curClipRect = parentLayer.clipRRect!.outerRect;
       } else if (parentLayer is ClipPathLayer) {
         curClipRect = parentLayer.clipPath!.getBounds();
+      } else if (parentLayer is PhysicalModelLayer) {
+        curClipRect = parentLayer.clipPath!.getBounds();
```

Comment:
I don't know whether this is the correct check, but I'd prefer combining these into a single `parentLayer is ClipPathLayer || parentLayer is PhysicalModelLayer` condition.